File: front/src/app/models/relatorioPonto.ts
import { PontoNormalizado } from "./pontoNormalizado";

export class RelatorioPonto {
    usuario: string;
    pontos: PontoNormalizado[];
    saldoString: string;
    pagamento: number;

    constructor(item?) {
        if (item) {
            this.usuario = item.usuario;
            this.pontos = item.pontos.map(x => new PontoNormalizado(x));
            this.saldoString = item.saldoString;
            this.pagamento = item.pagamento;
        }
    }
}

File: back/Infra/Views/RelatorioPonto.cs
using System;
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;

namespace SGE.Infra.Views
{
    [ExcludeFromCodeCoverage]
    public class RelatorioPonto
    {
        public string Funcionario { get; set; }

        public List<PontoNormalizado> Pontos { get; set; }

        public DateTimeOffset Saldo { get; set; }

        public string SaldoString { get; set; }
    }
}